data/campaigns/Descent_Into_Darkness/units/Mal_Keshar.cfg

```diff
@@ -0,0 +1,529 @@
+            apply_to=max_experience
+            increase={XP_INCREMENT_N}
+        [/effect]
+#endif
+#ifdef HARD
+        [effect]
+            apply_to=max_experience
+            increase={XP_INCREMENT_H}
+        [/effect]
+#endif
+        [effect]
+            apply_to=variation
+            name={VARIATION_NAME}
+        [/effect]
+        {EFFECT_WML}
+    [/advancement]
+#enddef
+
+############## LICH ADVANCEMENTS ##############
+
+#define LICH_MELEE VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 4 14 17 20 (
+        id=lich_melee
+        description= _ "+1 melee damage"
+        image=attacks/touch-undead.png
+        [effect]
+            apply_to=attack
+            name=touch
+            increase_damage=1
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define LICH_RANGED VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 2 22 26 30 (
+        id=lich_ranged
+        description= _ "+1 ranged damage"
+        image=attacks/dark-missile.png
+        [effect]
+            apply_to=attack
+            range=ranged
+            increase_damage=1
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define LICH_MOVEMENT VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 3 17 21 25 (
+        id=lich_movement_melee
+        description= _ "+1 movement"
+        image=icons/sandals.png
+        [effect]
+            apply_to=movement
+            increase=1
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+############## ANCIENT LICH ABILITIES ##############
+
+#define ANCIENT_LICH_MALICE VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 0 50 55 60 (
+        id=ancient_lich_malice
+        description= _ "Nearby undead units gain 20% damage"
+        image=attacks/wail.png
+        [effect]
+            apply_to=new_ability
+            [abilities]
+                {ABILITY_MALICE1}
+            [/abilities]
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_RESIST VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 0 70 75 80 (
+        id=ancient_lich_resist
+        description= _ "Gains +10% resistance to blade, impact, and pierce"
+        image=icons/potion_red_medium.png
+        [effect]
+            apply_to=resistance
+            [resistance]
+                blade=-10
+                impact=-10
+                pierce=-10
+            [/resistance]
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_NOVA VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 2 0 0 0 (
+        id=ancient_lich_nova
+        description= _ "Chill tempest gains area of effect damage"
+        image=attacks/iceball.png
+        [effect]
+            apply_to=attack
+            name=chill tempest
+            [set_specials]
+                {WEAPON_SPECIAL_MAGICAL}
+                {WEAPON_SPECIAL_FROST_NOVA}
+            [/set_specials]
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=-100%
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=89
```

Comment:
i think `set=` can alsobe used with max_expereince

Reply:
I don't think it's documented on the wiki, but it does seem to work. I replaced these blocks with set=.